services/versioncache/versioncache.go

```diff
@@ -0,0 +1,303 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package versioncache provides service software version cache functionality.
+package versioncache
+
+import (
+	"context"
+	"time"
+
+	"github.com/AlekSi/pointer"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+	"github.com/percona/pmm-managed/services/agents"
+)
+
+var (
+	serviceCheckInterval   = 24 * time.Hour
+	minCheckInterval       = 5 * time.Second
+	serviceFirstCheckDelay = 30 * time.Second
+)
+
+//go:generate mockery -name=Versioner -case=snake -inpkg -testonly
+
+// Versioner contains method for retrieving versions of different software.
+type Versioner interface {
+	GetVersions(pmmAgentID string, softwares []agents.Software) ([]agents.Version, error)
+}
+
+// Service is responsible for caching service software versions in the DB.
+type Service struct {
+	db       *reform.DB
+	l        *logrus.Entry
+	v        Versioner
+	updateCh chan struct{}
+}
+
+// New creates new service.
+func New(db *reform.DB, v Versioner) *Service {
+	return &Service{
+		db:       db,
+		l:        logrus.WithField("component", "version-cache"),
+		v:        v,
+		updateCh: make(chan struct{}, 1),
+	}
+}
+
+func (s *Service) syncServices() error {
```

Comment:
I don't think that we need this method at all.
We can create ServiceSoftwareVersions in the moment of migration using `insert into items_ver (item_id, name, item_group)
select item_id, name, item_group from items where item_id=2;
` construction and when we create a service. 
if we will remove ServiceSoftwareVersion for service the moment we delete service we won't have to do it here.

Reply:
syncServices removed, inserting on service insertion